tests/Console/ConfigurationResolverTest.php

```diff
@@ -246,11 +246,53 @@ public function testResolveConfigFileChooseFileWithInvalidFile()
     public function testResolvePathRelative()
     {
         $this->resolver
-            ->setCwd(__DIR__)
-            ->setOption('path', 'Foo'.DIRECTORY_SEPARATOR.'Bar')
+            ->setCwd(dirname(__DIR__))
+            ->setOption('path', basename(__DIR__))
             ->resolve();
 
-        $this->assertSame(__DIR__.DIRECTORY_SEPARATOR.'Foo'.DIRECTORY_SEPARATOR.'Bar', $this->resolver->getPath());
+        $this->assertSame(__DIR__, $this->resolver->getPath());
+    }
+
+    public function testResolvePathWithFileThatIsExcludedDirectly()
+    {
+        $dir = realpath(__DIR__);
+        $this->config->getFinder()
+            ->in($dir)
+            ->notPath(substr(__FILE__, strlen($dir) + 1));
+
+        $this->resolver
+            ->setOption('path', __FILE__)
+            ->resolve();
+
+        $this->assertCount(0, iterator_to_array($this->resolver->getConfig()->getFinder()));
+    }
+
+    public function testResolvePathWithFileThatIsExcludedByDir()
+    {
+        $dir = realpath(__DIR__.'/..');
+        $this->config->getFinder()
+            ->in($dir)
+            ->exclude(substr(realpath(__DIR__), strlen($dir) + 1));
```

Comment:
if `__DIR__would be in linked path then realpath will reveal real path that link point to.

If you have project forked in non-linked path then indeed, `$dir = realpath(__DIR__.'/..');` will only move one dir up and then second realpath is not needed. But if you would have in in linked path, then first realpath will reveal linked point, so the second one is needed.

If you have some nice way of normalizing `/foo/bar/.././baz` into `/foo/baz` I'm willing to listen, as it's the only reason I called realpath here.